test/TestProjects/MgmtDiscriminator/src/Generated/DeliveryRuleData.Serialization.cs

```diff
@@ -461,6 +477,33 @@ private BinaryData SerializeBicep(ModelReaderWriterOptions options)
                 }
             }
 
+            hasPropertyOverride = hasObjectOverride && propertyOverrides.TryGetValue("UnflattenedName", out propertyOverride);
+            if (Optional.IsDefined(Unflattened) || hasPropertyOverride)
```

Comment:
I feel like this if is redundant since its duplicated inside and can be removed.